apps/frontend/app/(website)/studio/src/store/snippets.ts

```diff
@@ -108,174 +107,186 @@ const getEditorsFromLS = () => {
-            console.error("error on JSON.stringify(obj.editors) ", { error });
-          }
-        };
-      },
-      setSelection: (editorsPairIndex, type) => (command) => {
-        const rootNode = get().editors[editorsPairIndex]?.[type]?.rootNode;
-        if (rootNode) {
-          const ranges = buildRanges(rootNode, command);
-
-          const obj = get();
-          obj.editors[editorsPairIndex][type].ranges = ranges;
-          obj.editors[editorsPairIndex][type].rangeUpdatedAt = Date.now();
-          set(obj);
-        }
-      },
-    }),
-    {
-      name: "snippets-storage",
-    },
-  ),
-);
+      100,
+    );
+  },
+  selectedPairIndex: 0,
+  getAllNames: () => get().editors.map(({ name }) => name),
+  getAllSnippets: () => {
+    return mapObjIndexed(
+      map(({ content }: SnippetValues) => content),
+      reduce(
+        (acc, { before, after, output }) => ({
+          before: [...acc.before, before],
+          after: [...acc.after, after],
+          output: [...acc.output, output],
+        }),
+        {
+          before: [],
+          after: [],
+          output: [],
+        } as AllEditors,
+        get().editors,
+      ),
+    );
+  },
+  setSelectedPairIndex: (i: number) => {
+    set({ selectedPairIndex: i });
+  },
+  getSelectedEditors: () => {
+    const index = get().selectedPairIndex || 0;
+    const editors = get().editors?.[index] as Editors;
+    return {
+      ...editors,
+      setContent: (type) => get().setContent(index, type),
+      beforeSnippet: editors?.before?.content || "",
+      afterSnippet: editors?.after?.content || "",
+      outputSnippet: editors?.output?.content || "",
+      setBeforeSnippet: get().setContent(index, "before"),
+      setAfterSnippet: get().setContent(index, "after"),
+      setOutputSnippet: get().setContent(index, "output"),
+      setBeforeSelection: get().setSelection(index, "before"),
+      setAfterSelection: get().setSelection(index, "after"),
+      setOutputSelection: get().setSelection(index, "output"),
+      setSelection: (editorType: EditorType) =>
+        get().setSelection(index, editorType),
+    };
+  },
+  setInitialState: (state) => {
+    set(state);
+  },
+  setEngine: (engine) => {
+    localStorage.setItem("engine", engine);
+    set({
+      engine,
+    });
+  },
+  setContent: (editorsPairIndex, type) => {
+    return (content) => {
+      const obj = get();
+      obj.editors[editorsPairIndex][type].content = content;
+      obj.editors[editorsPairIndex][type].rootNode = transformNode(
+        content,
+        type,
+      );
+      set({ currentContent: content, currentType: type, ...obj });
+      try {
+        localStorage.setItem(
+          "editors",
+          JSON.stringify(obj.editors.map(toEditorSnippets)),
+        );
+      } catch (error) {
+        console.error("error on JSON.stringify(obj.editors) ", { error });
+      }
+    };
+  },
+  setEditors: (editorsContents: EditorsSnippets[]) => {
```

Comment:
shorthand return

Reply:
no idea what do you mean - it is a store